esmvaltool/preprocessor/derived_variables/_derived_variable.py

```diff
@@ -0,0 +1,102 @@
+"""Contains the base class for derived variables."""
+
+
+import importlib
+
+
+class DerivedVariable(object):
+    """Base class for derived variables."""
+
+    def __init__(self, variable=None):
+        """Save `variable` dict and add `short_name` if necessary."""
+        if variable is None:
+            self.variable = {}
+        else:
+            self.variable = dict(variable)
+        if 'short_name' not in self.variable:
```

Comment:
This seems very unlikely to ever happen, because the only way to obtain a DerivedVariable subclass object is by knowing the 'short_name'.

Reply:
yes, this is case indeed (and we hit it not long ago) but then the damn netCDF file wouldn't `load` at all :grin: